Sample_v3/LoginViewController.swift

```diff
@@ -158,3 +150,13 @@ extension LoginViewController {
         }
     }
 }
+
+final class MyChatChannelListRouter: ChatChannelListRouter<DefaultExtraData> {
+    override func openChat(for channel: _ChatChannel<DefaultExtraData>) {
+        let chatStoryboard = UIStoryboard(name: "SimpleChat", bundle: nil)
+        let chatScreen = chatStoryboard
+            .instantiateViewController(withIdentifier: "SimpleChatViewController") as! SimpleChatViewController
+        chatScreen.channelController = rootViewController?.controller.client.channelController(for: channel.cid)
+        navigationController?.pushViewController(chatScreen, animated: true)
+    }
+}
```

Comment:
Why don't we introduce delegates for VCs instead of Routers?

Reply:
I'm actually thinking about the router as a `navigation delegate`. 

There's no much difference between `router.openCurrentUserProfile` and `delegate?.controller(self, didSelectCurrentUser:)`

Reply:
This is an interesting question we don't need to answer now. Let's stick with what we have here.